webapp/test/LoginForm.test.js

```diff
@@ -0,0 +1,107 @@
+import { mount, createLocalVue } from '@vue/test-utils';
+import Vuex from 'vuex';
+import LoginForm from '@/components/login-form/LoginForm';
+import {beforeEach, describe, expect, it} from "@jest/globals";
+import { state, getters } from '@/store';
+import Vuetify from 'vuetify'
+import Vue from 'vue'
+
+require('regenerator-runtime/runtime');
+
+// We create a scoped constructor that we can mutate.
+const localVue = createLocalVue();
+
+describe('LoginForm', () => {
+  let vuetify
+
+  let wrapper
+  let mutateMock
+  beforeEach(() => {
+    localVue.use(Vuex);
+    Vue.use(Vuetify);
+
+    vuetify = new Vuetify()
+
+    const mutations = {
+      setPrincipal: jest.fn((state, token) => state.user = { id: 1, token }),
+    }
+
+    const store = new Vuex.Store({
+      state,
+      mutations,
+      getters
+    });
+
+    mutateMock = jest.fn(( { mutation, variables } ) => {
+      return { data: { login: 'dummyToken' } };
+    })
+
+    wrapper = mount(LoginForm, {
+      localVue,
+      vuetify,
+      store,
+      mocks: {
+        $apollo: {
+          mutate: mutateMock
+        },
+        $apolloHelpers: {
+          onLogin: jest.fn()
+        }
+      }
+    });
+  });
+
+  it('keeps the login button disabled if the email is too long', async () => {
+    const email = 'testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest'
+    const submitButton = wrapper.find('#login');
+    const [emailExists, emailIsNotTooLong] = wrapper.vm.nameRules
```

Comment:
Consider `wrapper.vm` as private. If you want to unit test your validations, export them in your component:

```vue
<script>
export const validations = ...
export default {
  data() {
    validations,
    // ...
  }
}
</script>
```